src/org/jgroups/protocols/SSL_KEY_EXCHANGE.java

```diff
@@ -56,6 +58,57 @@ public interface SessionVerifier {
         void verify(SSLSession session) throws SecurityException;
     }
 
+    /**
+     * Helper class that knows how and when to reload the SSLContext.
+     */
+    protected class SSLContextReloader {
```

Comment:
can be made static

Reply:
Actually we are instantiating the reloader separately for the server and the client ssl context. The current (master) code allows the caller to supply either client or server context (or both), and any missing context is created on the fly. So we want to differentiate these cases. 
For example, if the caller supplies only the server context (client context created by SSL_KEY_EXCHANGE), then we only want to reload the client context. Caller is responsible to reload the server context.